nixos/tests/hibernate.nix

```diff
@@ -29,6 +30,11 @@ let
       "/".device = "/dev/vda2";
     };
     swapDevices = mkOverride 0 [ { device = "/dev/vda1"; } ];
+    boot.resumeDevice = mkIf systemdStage1 "/dev/vda1";
```

Comment:
Maybe instead of needing to set this, we can set it to the first viable swap device (i.e. `head resumeDevices`) along with an error or warning if there's more than one when stage-1-systemd is in use and `resumeDevice` isn't set.

Reply:
How does this currently work? The docs are vague:

> If left empty, the swap partitions are used.

Which swap partition? At what offset? How does it choose?

Reply:
It takes all of them. Uses the first one that identifies as a swap with a hibernated system. No offset. Yes, that's the current implementation

Reply:
Yikes, I think forcing the user to be explicit is better.

Reply:
We probably need some kind of message though. As is, enabling `initrd.systemd.enable` will silently disable hibernation if you don't have `resumeDevice` explicitly set.

Reply:
I also think most users don't expect to have to specify their resume device, particularly if they only have one swap device anyway. So this is a UX regression for *most* people IMO.